Session/Shared/FullConversationCell.swift

```diff
@@ -692,8 +696,8 @@ public final class FullConversationCell: UITableViewCell, SwipeActionOptimisticC
                 normalizedSnippet
                     .ranges(
                         of: (Singleton.hasAppContext && Singleton.appContext.isRTL ?
-                             "(\(part.lowercased()))(^|[^a-zA-Z0-9])" :
-                             "(^|[^a-zA-Z0-9])(\(part.lowercased()))"
+                             "(\(part.lowercased()))(^|[^a-zA-Z0-9])" : // stringlint:disable
+                             "(^|[^a-zA-Z0-9])(\(part.lowercased()))" // stringlint:disable
```

Comment:
This is a regex I mentioned... It is for conversation search and we may not be able to handle it perfectly :sob:

Reply:
Mm yea - I think we will need to do some testing (with both this one and the Mentions regex), I posted a bit on Discord around this but to add the content here:

So there are 2 main things that I'm not sure of in iOS:
**1)** When typing a message which includes a mention in a RTL language what order do the @ and {hex} end up in within the actual String variable
**2)** Does the regex logic process characters in different orders between RTL/LTR languages

I _think_ the RTL/LTR handling is more of a visual thing rather than a data thing (ie. the content in the String would still have characters ordered in a LTR way and the regex would process in an LTR way) but I don't **know** if that's the case 😅 

I'm hoping we can find a way to do a quick test to confirm the behaviour and, hopefully, ditch this `isRTL` check entirely (and not have to add one for detecting mentions)